### PR TITLE
Update and fix Bad User-Agent - Scanners

### PR DESCRIPTION
**Pull request by Rod-Gomes**

### CODE REVIEW

**Review by comotion — APPROVED** [review]

File: vcl/rules/generic.vcl [hunk 1]
```diff
@@ -4,7 +4,7 @@
 sub vcl_recv {
     # Bad User-Agent - Scanners
     # - http://mod-security.svn.sourceforge.net/ (modsecurity_35_scanners.data)
-    if (req.http.User-Agent ~ "(i)(metis|bilbo|n-stealth|black widow|brutus|cgichk|webtrends security|jaascois|pmafind|\.nasl|nsauditor|paros|nessus|nikto|webinspect|blackwidow)") {
+    if (req.http.User-Agent ~ "(?i:(?:^(grabber|bsqlbf|mozilla\/4\.0 (compatible)|sqlmap|mozilla\/4\.0 (compatible; msie 6.0; win32)|mozilla\/5\.0 sf\/\/|arachni|sql power injector|absinthe|netsparker|python-httplib2|dirbuster|pangolin|nmap nse|sqlninja|grendel-scan|havij|w3af|hydra|nstalker|n-stalker|openvas|fimap|yanga|url_spider_sql|topblogsinfo|purebot|jikespider|google_three_web|aboundexbot|mozilla\/the mole|themole\.nasel\.com\.ar|metis|bilbo|n-stealth|black widow|brutus|cgichk|webtrends security|jaascois|pmafind|\.nasl|nsauditor|paros|nessus|nikto|webinspect|blackwidow)$))") {
```
Comment (comotion) [inline vcl/rules/generic.vcl]: I'm curious if you've checked any of these - in particular, if you've seen the "mozilla/4.0 (compatible)" match? Parens are a special character in regex..

Comment (Rod-Gomes) [reply vcl/rules/generic.vcl]: @comotion You're right.

I actually copied this regex directly from the ModSecurity rules. This same rule is used globally by several hosting companies.

Despite incorrect parens, this should not affect the rule's functioning.
If you prefer, I can fix that.

Comment (comotion) [reply vcl/rules/generic.vcl]: yes, fix it please.

Comment (Rod-Gomes) [reply vcl/rules/generic.vcl]: fixed

**Review by Rod-Gomes — COMMENTED** [review]
fixed

**Comment by fgsch** [comment]
@comotion @Rod-Gomes @michbsd or anyone else using VSF, I appreciate this is 3 years old so I wonder whether this is still relevant. Thoughts?

**Comment by fgsch** [comment]
Actually, I will close this and update the list from OWASP CRS.

**Pull request closed without merging**